### PR TITLE
Decouple airship-setup-deployer from common socok8s vars

### PR DESCRIPTION
**Pull request by toabctl**
Try to keep roles as separated as possible. That makes the role
reusable also for non socok8s cases and the role has an interface (the
variables) to interact with.
Note: airship-setup-deployer still depends on the common-deployer
role.
x

### CODE REVIEW

**Comment by evrardjp** [comment]
I like the idea to decouple the role from the common vars, and inject those. :+1: 
on the principle.

I didnt' get the chance to dig into the code fully.

File: playbooks/roles/airship-setup-deployer/defaults/main.yml [hunk 1]
```diff
@@ -1,9 +1,19 @@
 ---
 # defaults file for suse-airship-deploy
 
+# IP on the Data Center Management (DCM) network
+dcm_vip: null
+
+# IP on the external network
+ext_vip: null
```
Comment (Itxaka) [inline playbooks/roles/airship-setup-deployer/defaults/main.yml]: do we need to add these? isnt that covered by the change above?

Comment (toabctl) [reply playbooks/roles/airship-setup-deployer/defaults/main.yml]: It is covered. But having it here is imo good to have the available variables documented.

Comment (toabctl) [reply playbooks/roles/airship-setup-deployer/defaults/main.yml]: @Itxaka is that answer ok? Or do you want me to change something here?

File: playbooks/generic-deploy_airship.yml [hunk 2]
```diff
@@ -2,8 +2,19 @@
 - hosts: soc-deployer
   gather_facts: yes
   any_errors_fatal: true
+  pre_tasks:
+    - name: Load generic vars
+      include_vars: "{{ item }}"
+      loop:
+        - "{{ playbook_dir }}/../vars/common-vars.yml"
+        - "{{ playbook_dir }}/../vars/deploy-on-openstack.yml"
```
Comment (kantarun) [inline playbooks/generic-deploy_airship.yml]: Not clear if there is any specific purpose of using vars as pre_tasks. If its just vars availability across roles, may be  vars_files will be more concise ?  https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#defining-variables-in-files

**Comment by toabctl** [comment]
> Just searching of this vip variable in socok8s repo, there are some more lingering references which needs to be updated as well

@kantarun no. I do not globally change the variable names. I'm only changing the names in the role `airship-setup-deployer` and make the role `airship-setup-deployer` independent from our global variable definitions .

File: playbooks/generic-deploy_airship.yml (hunk 2 above)
Comment (toabctl) [reply playbooks/generic-deploy_airship.yml]: Fixed. Thanks!

**Pull request merged**